packages/edit-site/src/store/selectors.js

```diff
@@ -170,3 +170,14 @@ export function isNavigationOpened( state ) {
 export function isInserterOpened( state ) {
 	return state.blockInserterPanel;
 }
+
+/**
+ * Returns the default template types definitions.
+ *
+ * @param {Object} state Global application state.
+ *
+ * @return {Object} The template types definitions.
+ */
+export function getDefaultTemplateTypesDefinitions( state ) {
+	return state.settings?.defaultTemplateTypesDefinitions;
```

Comment:
In the long run, do you think it should be exposed as another REST API endpoint?

Reply:
I don't see why not!
But also at the moment I don't see why yes. 😄 

Either way, with the list in PHP there's nothing that would technically prevent us to do so.